app/main/controller/presto_controller.py

```diff
@@ -0,0 +1,24 @@
+from flask import request, current_app as app
+from flask_restplus import Resource, Namespace
+
+from app.main.lib.fields import JsonObject
+from app.main.lib import similarity
+from app.main.lib.check_api import CheckAPI
+
+api = Namespace('presto', description='presto callback url')
+presto = api.model('presto', {
+    'body': JsonObject(required=False, description='Original body of the message sent - will contain ID, callback_url, and original url/text of request'),
+    'response': JsonObject(required=False, description='The response data sent back from the model'),
+})
+@api.route('/receive/<string:action>/<string:model_type>')
+class PrestoResource(Resource):
+    @api.response(200, 'Successfully responded to presto callback.')
+    @api.doc('Receive a presto callback for a given `model_type`')
+    def post(self, action, model_type):
+        data = request.args or request.json
+        if action == "add_item":
+            result = similarity.callback_add_item(data, model_type)
+            CheckAPI.return_storage_webhook(app.config['CHECK_API_HOST'], action, model_type, result)
```

Comment:
This makes me feel that this is a `checkapi_callback` endpoint not a `presto` one. In general it feels problematic to hardcode a callback to CheckAPI here. If Timpani wants to store an item we should be using all the same logic and just getting the callback URL from the initial call. Either that callback URL should be passed along the whole time or cached in Reddis by Alegre and looked up here.